charts/ceph-csi-cephfs/templates/nodeplugin-grpc-service.yaml

```diff
@@ -0,0 +1,41 @@
+{{- if .Values.nodeplugin.grpcMetrics.enabled -}}
+apiVersion: v1
+kind: Service
+metadata:
+{{- if .Values.nodeplugin.grpcMetrics.service.annotations }}
+  annotations:
+{{ toYaml .Values.nodeplugin.grpcMetrics.service.annotations | indent 4 }}
+{{- end }}
+  name: {{ include "ceph-csi-cephfs.nodeplugin.fullname" . }}-grpc-metrics
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ include "ceph-csi-cephfs.fullname" . }}
+    chart: {{ include "ceph-csi-cephfs.chart" . }}
+    component: {{ .Values.nodeplugin.name }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+{{- if .Values.nodeplugin.grpcMetrics.service.clusterIP }}
+  clusterIP: "{{ .Values.nodeplugin.grpcMetrics.service.clusterIP }}"
```

Comment:
this need to validate
cannot use cluster IP for node plugin I believe as this is because host networking is enabled

Reply:
I am testing this as we speak

Reply:
Just deployed the chart with:

```yaml
---
csiConfig:
  - clusterID: <redacted>
    monitors:
      - 192.168.2.100:6789

nodeplugin:
  httpMetrics:
    enabled: true
    service:
      type: ClusterIP

  grpcMetrics:
    enabled: true
    service:
      type: ClusterIP

provisioner:
  httpMetrics:
    enabled: true
    service:
      type: ClusterIP

  grpcMetrics:
    enabled: true
    service:
      type: ClusterIP
```

And I can curl cluster-ip:8090/metrics and cluster-ip:8080/metrics perfectly from within the cluster for both the nodeplugin and the provisioner :)

Reply:
Thanks for testing it